protecode/scanning_util.py

```diff
@@ -463,6 +558,129 @@ def wait_for_scan_to_finish(protecode_products_to_consider):
+                        extended_objects=component.extended_objects(),
+                    )
+
+    def process_component_resources(self) -> typing.Iterable[pm.BDBA_ScanResult]:
+        # depending on upload-mode, determine an upload-action for each related image
+        # - resources to upload
+        # - protecode-apps to remove
+        # - triages to import
+
+        logger.info(f'Processing component resource group for {self.component_name=} and '
+            + f'{self.resource_name=}')
+
+        metadata = self._image_group_metadata(
+            component_name=self.component_name,
+            resource_name=self.resource_name,
+        )
+
+        # Get all protecode appps
+        self.existing_protecode_products = self._api.list_apps(
+            group_id=self._group_id,
+            custom_attribs=metadata,
+        )
+
+        # for each protecode app get the scan results
+        scan_results = self._get_existing_protecode_apps()
+        logger.info('Found existing protecode apps:')
+        for r in scan_results:
+            logger.info(f'... {r.name()=}, {r.product_id()=}, {r.greatest_cve_score()=}')
+
+        # get triages for the scan results
+        triages_to_import: typing.Set[pm.Triage] = set()
+        triages_to_import |= set(self._existing_triages(scan_results))
+
+        # import triages from reference groups
+        triages_to_import |= set(self._enumerate_reference_triages(metadata))
+
+        logger.info(f'found {len(triages_to_import)} triage(s) to import')
+
+        # process resources according to processing mode
+        if self._processing_mode is ProcessingMode.FORCE_UPLOAD:
+            logger.info('force-upload - will re-upload all images')
+            self.component_resources_to_upload += list(self.component_resources)
+            # remove all
+            self.protecode_products_to_remove = set(self.existing_protecode_products)
+        elif self._processing_mode is ProcessingMode.RESCAN:
+            logger.info('rescan - will upload images to be rescanned')
+            self._get_for_rescan()
+        else:
+            raise NotImplementedError()
+
+        # trigger rescan if recommended for all protecode apps
+        self._trigger_rescan_if_recommended()
+
+        # upload new resources (all images to scan)
+        analysis_results = self._upload_new_resources()
+
+        # apply imported triages for all protecode products
+        self._apply_triages(analysis_results, triages_to_import)
+
+        # apply auto triages for resources labeled with skip binary scan
+        for protecode_product in self.protecode_products_to_consider:
+            component_resource = self.product_id_to_resource[protecode_product.product_id()]
+            has_skip_label = self._has_skip_label(component_resource.resource)
+            logger.info(f'{component_resource.component.name}  has skip-scan label {has_skip_label}')
+            if has_skip_label:
```

Comment:
```suggestion
            if self._has_skip_label(component_resource.resource):
```
Logging a bool here does not add much. Or are we interested that much in components without skip label? If so, I'd probably log in with an `else:` instead of logging twice in the positive case.

Reply:
Agreed, only logging if label is present.